projects/project-1/assignment/MMiroir - P1.md

```diff
@@ -0,0 +1,342 @@
+
+
+
+### Table II. Role Constraint Combinations***
+|            | Funct. | iFunct.| Trans. | Symm.     | Asymm.    | Ref.      |Irref.     |
+|--------    |--------|--------|--------|-------    |--------   |-----      |--------   |
+| **Funct.** |-       | OK     |X<sup>NS| OK        | OK        | OK        | OK        |
+| **iFunct.**|OK      | -      |X<sup>NS| OK        | OK        | OK        | OK        |
+| **Trans.** |X<sup>NS|X<sup>NS| -      | OK        |X<sup>NS   | OK        |X<sup>NS   |
+| **Symm.**  |OK      | OK     | OK     | -         |X<sup>UNSAT| OK        | OK        |
+| **Asymm.** |OK      | OK     |X<sup>NS|X<sup>UNSAT| -         |X<sup>UNSAT| OK        |
+| **Ref.**   |OK      | OK     | OK     |     OK    |X<sup>UNSAT| -         |X<sup>UNSAT|
+| **Irref.** | OK     | OK     |X<sup>NS|     OK    |    OK     |X<sup>UNSAT| -         |
+
+*_NS denotes a combination creates an object property chain that may lead to undecidability_
+
+**_UNSAT denotes a combination that is unsatisfiable_
+
+***_Provided by the assignment instructions_
+
+For the purposes of this assignment, the diagonal of the matrix in the above table (II) was ignored as these are not combinations of properties. Given that the matrix is also reflective across the diagonal, only side of the diagonal was observed and tested. Please note that this assignment does not explore Transitive & Inverse Functional (**C**) of the **NS** results or any of the **UNSAT** results.
+
+According to the table above (II), there are four property combinations which yielded an undecidable (**NS**) result: 
+
+- (**A**) Irreflexive **&** Transitive
+
+- (**B**) Transitive **&** Functional
+
+- (**C**) Transitive **&** Inverse Functional
+
+- (**D**) Transitive **&** Assymetric
+
+There are three property combinations which yield an unsatifiable (**UNSAT**) result:
+
+- (**E**) Assymetric **&** Symmetric
+
+- (**F**)) Assymetric **&** Reflexive 
+
+- (**G**) Reflexive **&** Irreflexive
+
+
+### Explanations
+
+The following example, provided in the assignment, serves as the guide for which all following explanations adhere to: 
+- To explain why `R` cannot be both asymmetric and symmetric, your explanation may take the form: Suppose `R` is both symmetric and asymmetric. Then by symmetry for any x and y, if x`R`y it follows that y`R`x. However, by asymmetry it also follows that it is not the case that y`R`x. Hence, `R` cannot be both symmetric and asymmetric. Similarly, to explain why `R` cannot be both transitive and inverse functional.
+
+#### (A) Transitive + Irreflexive (NS)
+- Suppose `R` is both irreflexive and transitive.  By irreflexivity, it is not the case that xRx. By transitivity, for any x and y, if x`R`y and y`R`z, it follows that x`R`z.
+
+When `R` is irreflexive it can never relate back to itself but when `R` is transitive, it can relate to multiple variables via the same relationship including but not necessarily limited to itself.
```

Comment:
The combination is non-simple, which leads to greater computational complexity, and so is forbidden in OWL 2. In this context, a ‘simple’ object property is one that has no direct or indirect sub-properties that are either transitive or defined using a chain of distinct object properties.